File: articles/grit-and-deliberate-practice/fortran/grit_deliberate_practice_score.f90
program grit_deliberate_practice_score
  implicit none
  real :: perseverance_effort, consistency_interests, grit
  real :: deliberate_practice, feedback_quality, coaching_access, practice_quality

  perseverance_effort = 0.82
  consistency_interests = 0.64
  deliberate_practice = 0.86
  feedback_quality = 0.61
  coaching_access = 0.55

  grit = 0.60 * perseverance_effort + 0.40 * consistency_interests
  practice_quality = 0.50 * deliberate_practice + 0.30 * feedback_quality + 0.20 * coaching_access

  print *, "Synthetic grit score:", grit
  print *, "Synthetic practice quality index:", practice_quality
end program grit_deliberate_practice_score

File: articles/grit-and-deliberate-practice/go/grit_deliberate_practice_score.go
package main

import "fmt"

func GritScore(perseveranceEffort, consistencyInterests float64) float64 {
	return 0.60*perseveranceEffort + 0.40*consistencyInterests
}

func PracticeQualityIndex(deliberatePractice, feedbackQuality, coachingAccess float64) float64 {
	return 0.50*deliberatePractice + 0.30*feedbackQuality + 0.20*coachingAccess
}

func main() {
	grit := GritScore(0.82, 0.64)
	practiceQuality := PracticeQualityIndex(0.86, 0.61, 0.55)

	fmt.Printf("Synthetic grit score: %.3f\n", grit)
	fmt.Printf("Synthetic practice quality index: %.3f\n", practiceQuality)
}
